src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -16,19 +21,37 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all properties whose names contain any of "
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+            + "Parameters: KEYWORD [MORE_KEYWORDS]..."
+            + "[" + PREFIX_TAG + "TAG]..."
+            + "\n"
+            + "Example: " + COMMAND_WORD + " alice bob charlie"
+            + PREFIX_TAG + "Condo "
+            + PREFIX_TAG + "4 rm ";
+
+    private final Predicate<Property> propertyPredicate;
+    private final NameContainsKeywordsPredicate namePredicate;
+    private final ContainsTagsPredicate tagsPredicate;
 
-    private final NameContainsKeywordsPredicate predicate;
+    public FindCommand(NameContainsKeywordsPredicate namePredicate) {
+        this(namePredicate, new ContainsTagsPredicate());
+    }
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    /**
+     * Creates a FindCommand that composes the given predicates.
+     *
+     * @param namePredicate Property predicate checking for name match.
+     * @param tagsPredicate Property predicate checking for containment of tags.
+     */
+    public FindCommand(NameContainsKeywordsPredicate namePredicate, ContainsTagsPredicate tagsPredicate) {
+        this.namePredicate = namePredicate;
+        this.tagsPredicate = tagsPredicate;
+        this.propertyPredicate = namePredicate.and(tagsPredicate);
     }
```

Comment:
I would think (and perhaps we should discuss) the main benefit of finding by tags is to allow the user to say find all 4rm flats, does the find function support that now?

Reply:
I also think the keyword should be optional, since the key point of this added functionality should be to enable "filtering by tags"